### PR TITLE
config: must set line terminator when use strict-format

### LINKED ISSUE #37338: [Lightning]: Panic on invalid Mertic countter operation
## Bug Report

Version: 6.1.0

Lightning panic on https://github.com/pingcap/tidb/blob/master/br/pkg/lightning/restore/restore.go#L2333

I think it's because float64(currOffset - startOffset) < 0.

```log
panic: counter cannot decrease in value

goroutine 202374 [running]:
github.com/prometheus/client_golang/prometheus.(*counter).Add(0x0?, 0xc069674c80?)
/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/counter.go:109 +0xf4
github.com/pingcap/tidb/br/pkg/lightning/restore.(*chunkRestore).deliverLoop(0xc0618805a0,
{0x445dbd8, 0xc0017c6040}

, 0xc06c80a3c0, 0xc0012af740, 0x3, 0xc002a8ff18?, 0xc002a8ffd0?, 0xc0016e5080)
/mnt/tidb/sql/br/pkg/lightning/restore/restore.go:2255 +0x7a3
github.com/pingcap/tidb/br/pkg/lightning/restore.(*chunkRestore).restore.func1()
/mnt/tidb/sql/br/pkg/lightning/restore/restore.go:2509 +0xab
created by github.com/pingcap/tidb/br/pkg/lightning/restore.(*chunkRestore).restore
/mnt/tidb/sql/br/pkg/lightning/restore/restore.go:2507 +0x31b
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #37338

Problem Summary:

### What changed and how does it work?

as title

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

manually check it from a oncall ticket

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/lightning/mydump/region_test.go [hunk 1]
```diff
@@ -487,3 +487,96 @@ func TestSplitLargeFileOnlyOneChunk(t *testing.T) {
...
+				Separator: ",",
+			},
+			StrictFormat:  true,
+			Filter:        []string{"*.*"},
+			MaxRegionSize: 2,
+		},
+	}
+	divideConfig := NewDataDivideConfig(cfg, 1, ioWorker, store, meta)
+
+	// in fact this is the wrong result, just to show the bug. pos mismatch with offsets
```
Comment (D3Hunter) [inline pkg/lightning/mydump/region_test.go:531]:
```suggestion
	// in fact this is the wrong result, just to show the bug. pos mismatch with offsets. and we might read more rows than expected
```

**Review by D3Hunter — APPROVED** [review]
or we can fix it by let parser read the longest terminator when terminator is not given explicitly, import into may have similar issue too

**Comment by lance6716** [comment]
> we can fix it by let parser read the longest terminator when terminator is not given explicitly

this fix has easier logic but require a user behaviour change, I'll ask PM's approval about this fix first. Change the parser behaviour also need to think about what's the position of last field in a line and line terminator in multiple cases, use `==` or `>=`, different lines may end with different terminator, etc. More complex.

> import into may have similar issue too

I also add the check for IMPORT INTO

**Comment by Frank945946** [comment]
@lance6716  Currently, \r, \n, and \r\n can all be used as line delimiters. However, when the user selects the Strict format, an additional check can be performed. This involves reading the first line of the source file to determine the line delimiter used in the source file, such as \r\n. If \r\n is identified as the line delimiter, it will be used to split one large CSV file into multiple smaller CSV files based on the specified size.

**Comment by lance6716** [comment]
> @lance6716 Currently, \r, \n, and \r\n can all be used as line delimiters. However, when the user selects the Strict format, an additional check can be performed. This involves reading the first line of the source file to determine the line delimiter used in the source file, such as \r\n. If \r\n is identified as the line delimiter, it will be used to split one large CSV file into multiple smaller CSV files based on the specified size.

There are many difficulties to implement the additional check. My arguments are

- considering the definition of "strict"-format, it's non-sense to tolerate `\r`, `\n`, `\r\n` all as terminator in one task or even in one file. The old behaviour in fact supports all three terminators in one file, and out detect logic will cause wrong parsing result.
- There are many execution path for this feature, like IMPORT INTO with/without DXF, IMPORT INTO running while upgrading, lightning resuming from checkpoint. Splitting the strict format files and parsing the file chunks are two separate procedures, and `strict-format` is not a sub-item of `csv` in configuration or in code. It's complecated and adding a restriction to user is less error-prone.

**Review by tangenta — APPROVED** [review]

**Pull request merged**